### PR TITLE
Validate number formatting locales

### PR DESCRIPTION
**Pull request by codewhisperer**

### CODE REVIEW

File: src/number/number.js [hunk 1]
```diff
@@ -31,6 +31,23 @@ function getPrecisionOptions(precision) {
...
 }
 
+/**
+ * Returns the locale that was passed in if it works
+ * with `toLocaleString` or otherwise falls back to `DEFAULT_LOCALE`
+ * @param {String} locale
+ */
+function getValidLocale(locale) {
+  try {
+    const value = 10;
```
Comment (OFFLlNE) [inline src/number/number.js]: Maybe would be nice to have a descriptive name here? What does this value indicate?

Comment (OFFLlNE) [reply src/number/number.js]: But while thinking it all through I kind of understand, so not sure if really necessary

Comment (codewhisperer) [reply src/number/number.js]: changed it `number`

File: src/number/number.js [hunk 2]
```diff
@@ -31,6 +31,23 @@ function getPrecisionOptions(precision) {
...
+ * Returns the locale that was passed in if it works
+ * with `toLocaleString` or otherwise falls back to `DEFAULT_LOCALE`
+ * @param {String} locale
+ */
+function getValidLocale(locale) {
+  try {
+    const number = 10;
+    const noUnderscoreLocale = locale.replace(/_/, '-');
+
+    number.toLocaleString(noUnderscoreLocale);
```
Comment (pterblgh) [inline src/number/number.js]: What about using `Intl.NumberFormat` here instead of `toLocaleString`?

image

**Pull request merged**